topdown/topdown_partial_test.go

```diff
@@ -2168,8 +2168,57 @@ func TestTopDownPartialEval(t *testing.T) {
 			wantSupport: []string{
 				`package partial.test
 
-				p = true { __local2__1 = input; __local2__1 = __local0__2; equal(__local0__2, 1, __local1__2); y2 = __local1__2; y2 }
-				`,
+				p = true { __local2__1 = input; data.partial.test.f(__local2__1) }
+				f(__local0__2) = y2 { equal(__local0__2, 1, __local1__2); y2 = __local1__2 }`,
+			},
+		},
+		{
+			note:    "shallow inlining: function with unknowns in rule body",
+			query:   "data.test.f(1, x)",
+			shallow: true,
+			modules: []string{
+				`package test
+				f(x) = true { input.x = x }
+				f(x) = false { input.y = x }`,
+			},
+			wantQueries: []string{`data.partial.test.f(1, x)`},
+			wantSupport: []string{
+				`package partial.test
+				f(__local0__2) = true { input.x = __local0__2 }
+				f(__local1__1) = false { input.y = __local1__1 }`,
+			},
+		},
+		{
+			note:    "shallow inlining: functions with no unknowns in rule body or output",
+			query:   "data.test.f(1, y)",
+			shallow: true,
+			modules: []string{
+				`package test
+				f(x) = true { x > 1 }
+				f(x) = false { x <= 0 }
+				f(x) = "meow" { false }`,
```

Comment:
ℹ️ This rule goes away: it doesn't depend on its arguments, and the body PE's to false.

The others, however, stay, since we're treating the function arguments as unknown during partial eval.